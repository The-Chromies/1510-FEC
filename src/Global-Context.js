/* eslint-disable react/destructuring-assignment */
/* eslint-disable react/prop-types */
/* eslint-disable prefer-const */
import React, { useState, createContext } from 'react';
import axios from 'axios';
import uuid from 'node-uuid';
import LocalServer from './env/localServ';

export const ContactContext = createContext();

export const ContactContextProvider = (props) => {
  // global state accessible to all components
  let [productId, setProductId] = useState(18085);
  const [revCount, setRevCount] = useState(0);
  const [avgRating, setAvgRating] = useState(0);
  // arr of products added to outfit list in rc
  const [outfitProduct, setOutfitProduct] = useState([]);
  // arr of styles added to outfit list in rc
  const [outfitStyle, setOutfitStyle] = useState([]);
  // selected product
  const [product, setProduct] = useState(null);
  // selected styles obj
  const [styles, setStyles] = useState(null);
  const localServer = LocalServer;

  // click tracking function
  const clickTracker = (widgetName, e) => {
  // post to /interactions endpoint w/ element's class, widget, & time of click
    let date = new Date();
    let clickData = {
      element: e.target.className,
      widget: widgetName,
      time: date.toTimeString(),
    };
    axios.post(`http://${localServer}:3000/interactions`, clickData)
      .then((res) => {
        // console.log('SUCCESSFUL POST TO INTERACTIONS');
        console.log(res);
      })
      .catch((err) => {
        // console.log('FAILED TO POST TO INTERACTIONS');
        console.log(err);
      });
  };

  const generateStarImage = (ratingValue, keyId) => {
    const starCount = ratingValue > 0 ? (Math.round((ratingValue) * 4) / 4).toFixed(2) : 0;

    let remainder = 0;
    remainder = starCount - remainder;
    const starArr = [];
    for (let j = 0; j < 5; j += 1) {
      switch (remainder) { // 4.25
        case 0.25:
          starArr.push(<div key={uuid()}><img src="public/icons/star/quarterStar.png" alt="" className="star-image" /></div>);
          remainder -= 0.25;
          break;
        case 0.50:
          starArr.push(<div key={uuid()}><img src="public/icons/star/halfStar.png" alt="" className="star-image" /></div>);
          remainder -= 0.50;
          break;
        case 0.75:
          starArr.push(<div key={uuid()}><img src="public/icons/star/threeQuarterStar.png" alt="" className="star-image" /></div>);
          remainder -= 0.75;
          break;
        case 1:
          starArr.push(<div key={uuid()}><img src="public/icons/star/fullStar.png" alt="" className="star-image" /></div>);
          remainder -= 1;
          break;
        case 0:
          starArr.push(<div key={uuid()}><img src="public/icons/star/emptyStar.png" alt="" className="star-image" /></div>);
          break;
        default:
          starArr.push(<div key={uuid()}><img src="public/icons/star/fullStar.png" alt="" className="star-image" /></div>);
          remainder -= 1;
          break;
      }
    }
    return (<div className="inline-star" style={{ display: 'inline-flex' }} key={uuid()}>{starArr}</div>);
  };

  // get selected product & style for ov & rc components
  const getProduct = (id) => {
    axios.get(`http://${localServer}:3000/related/product/${id}`)
      .then((results) => {
        console.log('OUTFITPRODUCT!!', results.data);
        let outfitProductList = [...outfitProduct];
        outfitProductList.push(results.data);
        setOutfitProduct(outfitProductList);
      })
      .catch((err) => {
        console.log('err getting outfit product info:', err);
      });
  };

  const getStyle = (id) => {
    axios.get(`http://${localServer}:3000/overview/styles/${id}`)
      .then((results) => {
        let outfitStyleList = [...outfitStyle];
        outfitStyleList.push(results.data);
        setOutfitStyle(outfitStyleList);
      });
  };

  // add product to outfit list
  const handelAddOutfit = (id) => {
    setProductId(id);
    getProduct(id);
    getStyle(id);
  };

  return (
    <ContactContext.Provider value={{
      productId,
      setProductId,
      generateStarImage,
      revCount,
      setRevCount,
      avgRating,
      product,
      styles,
      setAvgRating,
      clickTracker,
      outfitProduct,
      setOutfitProduct,
      outfitStyle,
      setOutfitStyle,
      getProduct,
      getStyle,
      handelAddOutfit,
      localServer,
    }}
    >
      {props.children}
    </ContactContext.Provider>
  );
};
