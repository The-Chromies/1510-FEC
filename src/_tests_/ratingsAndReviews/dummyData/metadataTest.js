const metaTest = {
  product_id: '18445',
  ratings:
    {
      1: '7', 2: '2', 3: '3', 4: '2', 5: '3',
    },
  recommended:
        { false: '7', true: '10' },
  characteristics:
        {
          Size: {
            id: 61906,
            value: '2.9411764705882353',
          },
          Width: {
            id: 61907,
            value: '2.9411764705882353',
          },
          Comfort: {
            id: 61908,
            value: '2.2941176470588235',
          },
          Quality: {
            id: 61909,
            value: '2.7647058823529412',
          },
        },
};

export default metaTest;
