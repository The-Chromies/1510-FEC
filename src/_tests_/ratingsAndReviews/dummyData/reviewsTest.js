const reviewsTest = {
  product: '18085',
  page: 0,
  count: 2,
  results: [{
    review_id: 288583, rating: 3, summary: 'fffeee', recommend: false, response: null, body: 'daafsesff', date: '2021-03-09T00:00:00.000Z', reviewer_name: 'Taylor Smart', helpfulness: 0, photos: [],
  }, {
    review_id: 288501, rating: 5, summary: 'Yeezy the best', recommend: true, response: null, body: 'Because ye is the best', date: '2021-03-09T00:00:00.000Z', reviewer_name: 'Ye', helpfulness: 0, photos: [],
  }],
};

export default reviewsTest;
