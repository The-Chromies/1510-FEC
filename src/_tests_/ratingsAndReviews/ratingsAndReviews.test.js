/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable no-undef */
import React from 'react';
import Enzyme, {
  configure, shallow, mount, render,
} from 'enzyme';
import renderer from 'react-test-renderer';
import RatingsAndReviews from '../../components/ratingsAndReviews/rrMain';
import ReviewBox from '../../components/ratingsAndReviews/components/reviewBox';
import ReviewFilter from '../../components/ratingsAndReviews/components/reviewFilter';
import ReviewListContainer from '../../components/ratingsAndReviews/components/reviewListContainer';
import SubmitReview from '../../components/ratingsAndReviews/components/submitReview';
import SummaryContainer from '../../components/ratingsAndReviews/components/summaryContainer';
import SummaryStar from '../../components/ratingsAndReviews/components/summaryStar';

import reviewsTest from './dummyData/reviewsTest';
import metadataTest from './dummyData/metadataTest';
import { ContactContextProvider } from '../../Global-Context';

describe('rrMain Tests', () => {
  let wrapper;
  const dummyprops = {
    revFlag: true,
    findReviewMeta: () => {},
    meta: [],
    generateStarImage: () => {},
    avgRating: 1,
    handleStarClick: () => {},
    starFilter: 1,
    handleFetchMore: () => {},
    revCount: 1,
    reviewList: [],
    productId: 18085,
  };
  it('Container is present when review list ', () => {
    const reviewList = reviewsTest;
    wrapper = mount(
      <ContactContextProvider>
        <RatingsAndReviews reviewList={reviewList} productId={0} />
      </ContactContextProvider>,
    );
  });

  // console.log(wrapper.debug())
  // expect(wrapper.find('.review-key-container')).toBeUndefined();

  it('Review Keys Container Present', () => {
    expect(wrapper.find('.review-key-container')).toBeTruthy();
  });

  it('Review Container Present', () => {
    expect(wrapper.find('.review-container')).toBeTruthy();
  });

  it('Summary Container Present', () => {
    expect(wrapper.find('.summary-container')).toBeTruthy();
  });

  it('review-box Present', () => {
    expect(wrapper.find('.review-box')).toBeTruthy();
  });

  // it('SummaryContainer React Component Present', () => {
  //   expect(wrapper.shallow(<SummaryContainer {...dummyprops} />).toBe(true));
  // });

  // it('ReviewFilter React Component Present', () => {
  //   expect(wrapper.shallow(<ReviewFilter {...dummyprops} />).toBe(true));
  // });

  // it('ReviewListContainer React Component Present', () => {
  //   expect(wrapper.shallow(<ReviewListContainer {...dummyprops} />).toBe(true));
  // });
});

describe('Compare Snapshot for each component', () => {
  const formValues = Object.freeze({
    product_id: 18055,
    rating: 1,
    summary: '',
    body: '',
    recommend: false,
    name: '',
    email: '',
    photos: [],
    characteristics: {},
    helpfulness: 1,
  });
  const dummyprops = {
    revFlag: true,
    findReviewMeta: () => {},
    meta: [{ ratings: [1, 2, 3, 4] }],
    generateStarImage: () => {},
    avgRating: 1,
    handleStarClick: () => {},
    starFilter: 1,
    handleFetchMore: () => {},
    revCount: 1,
    reviewList: [],
    productId: 18085,
    review: formValues,
    tempKey: 1,
  };
  it('matches the rrMain snapshot', () => {
    const snapshot = renderer
      .create(<ContactContextProvider><RatingsAndReviews {...dummyprops} /></ContactContextProvider>)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  it('matches the ReviewBox snapshot', () => {
    const snapshot = renderer
      .create(<ReviewBox {...dummyprops} />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  it('matches the ReviewFilter snapshot', () => {
    const snapshot = renderer
      .create(<ReviewFilter {...dummyprops} />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  it('matches the ReviewListContainer snapshot', () => {
    const snapshot = renderer
      .create(<ReviewListContainer {...dummyprops} />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  it('matches the SummaryContainer snapshot', () => {
    const snapshot = renderer
      .create(<SummaryContainer {...dummyprops} />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  it('matches the SubmitReview snapshot', () => {
    const snapshot = renderer
      .create(<SubmitReview {...dummyprops} />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
  it('matches the SummaryStar snapshot', () => {
    const snapshot = renderer
      .create(<SummaryStar {...dummyprops} />)
      .toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});

describe('Filter Container Check', () => {
  const reviewMeta = metadataTest;
  const setSortKey = () => {};
  it('ReviewFilter has default state value', () => {
    const wrapper = mount(<ReviewFilter meta={reviewMeta} setSortKey={setSortKey} />);
    const select = wrapper.find('select');
    expect(select.props().value).toBeTruthy();
  });
});

// it('renders list-items', () => {
//   const items = ['one', 'two', 'three'];
//   const wrapper = shallow(<List items={items} />);

//   // Expect the wrapper object to be defined
//   expect(wrapper.find('.list-items')).toBeDefined();
//   expect(wrapper.find('.item')).toHaveLength(items.length);
// });
