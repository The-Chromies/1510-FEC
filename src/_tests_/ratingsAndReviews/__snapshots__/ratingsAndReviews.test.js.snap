// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare Snapshot for each component matches the ReviewBox snapshot 1`] = `
<div
  className="review-box border border-secondary p-2 mb-3 rounded shadow"
>
  <div
    className="flex-row row"
  >
    <div
      className="review-rating justify-content-start col-md-4 col-12"
    >
      <span />
    </div>
    <div
      className="review-profile text-uppercase font-weight-bold col-sm col"
    >
      <span />
    </div>
    <div
      className="review-profile text-muted font-weight-light justify-content-end col-sm col"
    >
      <time>
        NaN-NaN-NaN
      </time>
    </div>
  </div>
  <div
    className="review-content"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h3
          className="review-title  text-bolder text-truncate"
        >
          
        </h3>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <span
          className="review-body text-body border-info lead"
        >
          <div>
            
          </div>
        </span>
      </div>
    </div>
  </div>
  <div
    className="secondary-meters"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <button
          className="btn-sm btn-success"
          id="rev-help"
          onClick={[Function]}
          type="button"
        >
          Helpfulness - 1
        </button>
      </div>
      <div
        className="col"
      >
        <button
          className="btn-sm btn-outline-info"
          id="rev-report"
          onClick={[Function]}
          type="button"
        >
          Report
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Compare Snapshot for each component matches the ReviewFilter snapshot 1`] = `
<div
  className="review-filter flex-fill d-flex align-content-center align-items-center justify-content-center"
>
  <div
    className="row"
  >
    <p
      className="text-center font-italic pt-3 pr-2"
    >
      undefined reviews, sorted by  
    </p>
    <select
      className="select-review-filter"
      onClick={[Function]}
    >
      <option
        value="relevant"
      >
        Relevance
      </option>
      <option
        value="helpful"
      >
        Helpful
      </option>
      <option
        value="newest"
      >
        Newest
      </option>
    </select>
  </div>
</div>
`;

exports[`Compare Snapshot for each component matches the ReviewListContainer snapshot 1`] = `
<div
  className="container review-container"
>
  <button
    className="btn-outline-light btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Add Review
  </button>
  <button
    className="btn-outline-light btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Load More
  </button>
</div>
`;

exports[`Compare Snapshot for each component matches the SubmitReview snapshot 1`] = `null`;

exports[`Compare Snapshot for each component matches the SummaryStar snapshot 1`] = `
<div
  className="row"
>
  <div
    className="justify-content-end font-weight-bold col-md-4 col-4"
  >
     
    0
     
  </div>
  <div
    className="col-md-8 col-8"
  >
     
     
  </div>
</div>
`;

exports[`Compare Snapshot for each component matches the rrMain snapshot 1`] = `null`;
